src/hash/clu_alg_hash.c

```diff
@@ -48,10 +67,31 @@ int wolfCLU_md5Setup(int argc, char** argv)
         }
     }
 
+    /* sets default size of algorithm */
+#ifndef NO_MD5
+    if (XSTRNCMP(alg, "md5", 3) == 0)
+        size = WC_MD5_DIGEST_SIZE;
+#endif
+
+#ifndef NO_SHA256
+    if (XSTRNCMP(alg, "sha256", 6) == 0)
+        size = WC_SHA256_DIGEST_SIZE;
+#endif
+
+#ifdef WOLFSSL_SHA384
+    if (XSTRNCMP(alg, "sha384", 6) == 0)
+        size = WC_SHA384_DIGEST_SIZE;
+#endif
+
+#ifdef WOLFSSL_SHA512
+    if (XSTRNCMP(alg, "sha512", 6) == 0)
+        size = WC_SHA512_DIGEST_SIZE;
+#endif
+    
     /* hashing function */
-    ret = wolfCLU_hash(bioIn, bioOut, (char*)"md5", WC_MD5_DIGEST_SIZE);
+    ret = wolfCLU_hash(bioIn, bioOut, alg, size);
     wolfSSL_BIO_free(bioIn);
-#endif
+    
```

Comment:
added in whit space at the end of the line here, ping me if using VIM and I can give you a way to view the trailing white spaces

Reply:
I looked it up and found `:set list` for vim. I've been using a 3rd app this entire time lol